client/src/modules/cohort/model/actions.js

```diff
@@ -106,3 +198,92 @@ export const updateCohort = (cohortId, data) => dispatch => {
       );
     });
 };
+
+export const deleteCohort = cohortId => dispatch => {
+  dispatch(deleteCohortRequest());
+  return deleteData(`${ENDPOINT_URL}/cohorts/${cohortId}/delete`)
+    .then(resp =>
+      resp.json().then(json => {
```

Comment:
can you separate this then as in below method archiveCorot. 
First return resp.json().
And then perform .then in next line